docs/development/doc_guide.rst

```diff
@@ -1,76 +1,633 @@
+  `paywall <https://en.wikipedia.org/wiki/Paywall>`_.
+
+.. note::
+
+   Emphasize important points with `admonitions
+   <https://docutils.sourceforge.io/docs/ref/rst/directives.html#admonitions>`_
+   like this one.
+
+Docstring guidelines
+--------------------
+
+* All functions, classes, and objects that are part of PlasmaPy's
+  public Application Programming Interface (API) must have a docstring
+  that follows the `numpydoc`_ standard.
+
+* The first line of the docstring for a function or method should begin
+  with a word like "Return", "Calculate", or "Compute" and end with a
+  period.
+
+* The first line of an object that is not callable (for example, an
+  attribute of a class decorated with `property`) should not begin with
+  a verb and should end with a period.
+
+* Keep the docstring indented at the same level as the ``r"""`` or
+  ``"""`` that begins the docstring, except for reST constructs like
+  lists, math, and code blocks. The indentation level should be four
+  spaces more than the declaration of the object.
+
+  .. code-block:: python
+
+     def some_function():
+         """This is indented four spaces relative to the `def` statement."""
+
+* The first sentence of a docstring of a function should include a
+  concise definition of the quantity being calculated, as in the
+  following example.
+
+  .. code-block:: python
+
+     def beta(T, n, B):
+         """Compute the ratio of thermal pressure to magnetic pressure."""
+
+  When the definition of the quantity being calculated is unable to fit
+  on ∼1–2 lines, include the definition in the extended summary instead.
+
+  .. code-block:: python
+
+     def beta(T, n, B):
+         """
+         Compute plasma beta.
+
+         Plasma beta is the ratio of thermal pressure to magnetic pressure.
+         """
+
+* Put any necessary highly technical information in the "Notes" section
+  of a docstring.
+
+* Private code objects (e.g., code objects that begin with a single
+  underscore) should have docstrings. A docstring for a private code
+  object may be a single line, and otherwise should be in `numpydoc`_
+  format.
+
+* Dunder methods (e.g., code objects like ``__init__`` that begin and
+  end with two underscores) only need to have docstrings when needed to
+  describe non-standard or potentially unexpected behavior.
+
+* When an attribute in a class has both a getter (which is decorated
+  with `property`) and a setter method, both the getter and setter
+  should be documented in the docstring for the getter.
+
+  .. code-block:: python
+
+     class Person:
+         @property
+         def age(self):
+             """Document both getter and setter here."""
+             return self._age
+
+         @age.setter
+         def age(self, n):
+             self._age = n
+
+Narrative documentation guidelines
+----------------------------------
+
+* Each subpackage in PlasmaPy must have corresponding narrative
+  documentation.
```

Comment:
```suggestion
* Each top-level subpackage in PlasmaPy must have corresponding narrative
  documentation.
```

I don't think this is even currently true for all sub-subpackages, and I'm okay with that.